src/matter/session/SessionManager.ts

```diff
@@ -59,11 +60,38 @@ export class SessionManager<ContextT> {
         return this.sessions.get(sessionId);
     }
 
+    destroySession(sessionId: number) {
+        const session = this.sessions.get(sessionId);
+        if (session) {
+            session.destroy();
+            this.sessions.delete(sessionId);
+        }
+    }
+
+    destroySessionForFabricIndex(fabricIndex: FabricIndex) {
```

Comment:
This should done be the FabricManager, add a getSessionsForFabric here and the FabricManager will be in charge of calling destroySession. Each manager should take of his own mess...